### PR TITLE
Removes unnecessary `func ==()` implementations and a few Equatable(s) 

### LINKED ISSUE #4342: Some func == () implementations are unnecessary

### PR DESCRIPTION
**Pull request by hboon**
Closes #4342

### CODE REVIEW

File: AlphaWalletTests/Tokens/AddressAndRPCServerTests.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import XCTest
+@testable import AlphaWallet
+
+class AddressAndRPCServerTests: XCTestCase {
+    func testEquality() {
+        let address1 = AlphaWallet.Address(string: "0x007bee82bdD9e866B2bD114780A47F2261C6840E")!
+        let address2 = AlphaWallet.Address(string: "0x007bee82bdD9e866B2bD114780A47F2261C6840e")!
+        let address3 = AlphaWallet.Address(string: "0xA66A3F08068174e8F005112A8b2c7A507a822335")!
+        XCTAssert(AddressAndRPCServer(address: address1, server: .main) == AddressAndRPCServer(address: address2, server: .main))
+        XCTAssert(AddressAndRPCServer(address: address1, server: .rinkeby) != AddressAndRPCServer(address: address2, server: .main))
```
Comment (eviltofu) [inline AlphaWalletTests/Tokens/AddressAndRPCServerTests.swift:12]: Why does this pass? Is the .rinkeby server case sensitive?

**Review by eviltofu — APPROVED** [review]

File: AlphaWalletTests/Tokens/AddressAndRPCServerTests.swift (hunk 1 above)
Comment (hboon) [reply AlphaWalletTests/Tokens/AddressAndRPCServerTests.swift:12]: I don’t understand. Can you rephrase?

Comment (hboon) [reply AlphaWalletTests/Tokens/AddressAndRPCServerTests.swift:12]:
```
let address1 = AlphaWallet.Address(string: "0x007bee82bdD9e866B2bD114780A47F2261C6840E")!
let address2 = AlphaWallet.Address(string: "0x007bee82bdD9e866B2bD114780A47F2261C6840e")!
let address3 = AlphaWallet.Address(string: "0xA66A3F08068174e8F005112A8b2c7A507a822335")!
XCTAssert(AddressAndRPCServer(address: address1, server: .main) == AddressAndRPCServer(address: address2, server: .main))
XCTAssert(AddressAndRPCServer(address: address1, server: .rinkeby) != AddressAndRPCServer(address: address2, server: .main))
```

> Why does this pass? Is the .rinkeby server case sensitive?

But if you are referring to why address1 and 2 are equivalent, yes, addresses aren't case-sensitive, only the EIP55 form is, but it's still the same address. Both address1 and address2's EIP55 form is `0x007bee82bdD9e866B2bD114780A47F2261C6840E` (uppercase E at the end). Nothing to do with chains.

But for `AddressAndRPCServer`, equality means the address are equal + same chain
